Forta-Bots-Challenge-3/src/utils.ts

```diff
@@ -0,0 +1,96 @@
+import { Finding, getAlerts, sendAlerts } from "forta-agent";
+import { providers, Contract, BigNumber } from "ethers";
+import { createEscrowBalanceFinding, createFinding } from "./finding";
+import { agentId } from "../forta.config.json";
+import { ERC20_ABI } from "./constants";
+
+const chainNames = {
+  1: "Ethereum",
+  10: "Optimism",
+  42161: "Arbitrum",
+};
+
+export type contractAddresses = {
+  l1DaiAddress: string;
+  l2DaiAddress: string;
+  optimismEscrow: string;
+  arbitrumEscrow: string;
+};
+
+export const emitEscrowBalance = async (
+  provider: providers.Provider,
+  blockNumber: number,
+  l1DaiAddress: string,
+  optimismEscrow: string,
+  arbitrumEscrow: string
+) => {
+  const l1DaiContract = new Contract(l1DaiAddress, ERC20_ABI, provider);
+
+  // call L1 DAI contract to get balances of each L2's escrow contract
+  const optimismEscrowBalance = await l1DaiContract.balanceOf(optimismEscrow, { blockTag: blockNumber });
+  const arbitrumEscrowBalance = await l1DaiContract.balanceOf(arbitrumEscrow, { blockTag: blockNumber });
+
+  // create finding
```

Comment:
An optimization suggestion would be to create an alert on this side only when the balance has changed from the previous alert